util/util.js

```diff
@@ -45,6 +45,10 @@ function getPreferredBName(hostObject) {
     return null;
   }
 
+  if (hostObject.name) {
```

Comment:
".name" is a special key for user customized name, should not be included in getPreferredBName function.